### PR TITLE
[KYUUBI #1090] Add deployment document about Hadoop Credentials Manager

### PR DESCRIPTION
**Pull request by zhouyifan279**
### _Why are the changes needed?_
Umbrella issue #915 is finished.  Document is needed to tell user how to enable it.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zhouyifan279** [comment]
@yaooqinn @ulysses-you ready to be reviewed.

File: docs/security/proxyuser.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+kyuubi\.credentials<br>\.hive\.enabled|<div style='width: 65pt;word-wrap: break-word;white-space: normal'>true</div>|<div style='width: 170pt;word-wrap: break-word;white-space: normal'>Whether to renew Hive metastore delegation token</div>|<div style='width: 30pt'>boolean</div>|<div style='width: 20pt'>1.4.0</div>
+kyuubi\.credentials<br>\.renewal\.interval|<div style='width: 65pt;word-wrap: break-word;white-space: normal'>PT1H</div>|<div style='width: 170pt;word-wrap: break-word;white-space: normal'>How often Kyuubi renews one user's delegation tokens</div>|<div style='width: 30pt'>duration</div>|<div style='width: 20pt'>1.4.0</div>
+kyuubi\.credentials<br>\.renewal\.retryWait|<div style='width: 65pt;word-wrap: break-word;white-space: normal'>PT1M</div>|<div style='width: 170pt;word-wrap: break-word;white-space: normal'>How long to wait before retrying to fetch new credentials after a failure.</div>|<div style='width: 30pt'>duration</div>|<div style='width: 20pt'>1.4.0</div>
```
Comment (yaooqinn) [inline docs/security/proxyuser.md:69]: can we rename retryWait to retry.wait? then users do not need to worry about case sensitivities

Comment (zhouyifan279) [reply docs/security/proxyuser.md]: OK. Renamed.

File: docs/security/proxyuser.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+<div align=center>
+
+![](../imgs/kyuubi_logo.png)
+
+</div>
+
+# Hadoop Credentials Manager
+
+Kyuubi engine is submitted with SparkSubmit argument `--proxy-user`. Its delegation tokens of hadoop
```
Comment (yaooqinn) [inline docs/security/proxyuser.md:26]: What if the engine is launched via keytab?

File: docs/security/index.rst [hunk 3]
```diff
@@ -25,5 +25,6 @@ Kyuubi Security Overview
 
     Authentication <authentication>
     kinit
+    proxyuser
```
Comment (yaooqinn) [inline docs/security/index.rst:28]: proxyuser seems not to be relevant?

Comment (zhouyifan279) [reply docs/security/index.rst]: Renamed it to hadoop_credentials_manager.md

File: docs/security/proxyuser.md (hunk 2 above)
Comment (zhouyifan279) [reply docs/security/proxyuser.md]: Added description for engine submitted with `spark.kerberos.principal` and `spark.kerberos.keytab`.

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**